### DISCUSSION TITLE
failed restart or recover from a checkpoint on HPC cluster

### DISCUSSION THREAD

**Discussion by xajezade** [idaholab/moose#26805] [Q&A Modules: Solid mechanics]
Hello there,

I have done a crystal plasticity simulation with contact mechanics and my simulation was killed due to time limit. Now when I want to restart or recover, I get segmentation fault. It looks like that the restart is loaded. I am using `--n-threads=32` and replicated mesh.  

Could you please help me to figure this out?
Ali

**Comment by GiudGiud** [comment]
Hello

DId you output one or two checkpoint files? Can you try the older one? If the simulation is killed while the file is being written, it will be corrupted.
Does it load fine without threads?

Guillaume

**Reply by schakrabortygithub** [reply]
This is a known issue #26485, specific to the crystal plasticity part of the tensor_mechanics module. Updates are not there in the main code yet.

**Reply by xajezade** [reply]
Hi
@GiudGiud Thank you for your reply! I have outputted two checkpoints, and non of them work. It doesn't work with single thread. It is N to N kind of restart.
@schakrabortygithub Ok, thank you! Can we keep this conversation open until the updates are merged into the main?

**Reply by GiudGiud** [reply]
It s ok to keep the post open. It s best to subscribe to the relevant issue notifications to stay informed about the topic btw

**Reply by xajezade** [reply]
@schakrabortygithub I have reinstalled MOOSE following the merged resolved issue. Now I can see that the recovery option works but my time increments quickly goes to a very small time increment. my time increment was stable at dt = 0.0004 before previous job termination but now it goes to  dt = 3.8147e-10 after 10 steps. Is this an expected behavior? Probably I'll hit "dt min reached" error!

**Reply by schakrabortygithub** [reply]
@xajezade I do not know the input script of your crystal plasticity simulation. Is it possible to share the minimal input script that will reproduce the problem.
Otherwise, can you can first try with some simple CP problem by using the script located at `modules/solid_mechanics/test/tests/crystal_plasticity/stress_update_material_based/bicrystal_test.i` and then restart the simulation. Plot the `time vs pk2` from both the simulations in one graph. You should get something similar to this:

![Stress_Vs_Time](https://github.com/idaholab/moose/assets/35871741/8a3e982f-56dc-49d4-bd7f-e3563896a632)

**Reply by GiudGiud** [reply]
The time step reducing means the solve is failing. 

See the troubleshooting failed solves instructions to get it to converge

**Reply by xajezade** [reply]
@schakrabortygithub I am going to test this! I am simulating indentation problem, so I have also contact in addition to crystal plasticity. I'll try to make a simpler model to see if I can reproduce the same issue for further investigation.

@GiudGiud I have an interrupted simulation due to the time limit that I have on our HPC system. Before the simulation was killed, I had a stable time increment and a converged solution. I have outputted the last two checkpoints, I get the same problem if I restart from both. Doesn't it suggest that the restart procedure caused this failing?

**Reply by GiudGiud** [reply]
If you can reproduce this failing on even an early checkpoint, a time step which would have passed if you did not use checkpoints, then yes it points to a failure in checkpoints

Which moose commit are you using now? It should say in the header of the log

**Reply by xajezade** [reply]
Here is my screenshot of the simulation log on the time increments before restart:
![TimeIncremenBeforeRestart](https://github.com/idaholab/moose/assets/30939214/20ae89b3-2f4f-4ce8-8b1a-5151ff93ac54)
and here is just after restart:
![TimeIncremenRestartHead](https://github.com/idaholab/moose/assets/30939214/d7245780-d9fa-4c5a-a087-57196ee32610)
Again my simulation is still running but with a very small time increment:
![TimeIncremenRestartTail](https://github.com/idaholab/moose/assets/30939214/bbb0761b-8956-4b77-9c00-a00a066bf4a9)

It looks like that the restart cannot restore contact properly and want to reestablish contact. Note that I don't have a problem without contact. I am going to test the example on 2d indenter to see if I can reproduce the same issue.
Here is the Framework information:
Framework Information:
MOOSE Version:           git commit bfbeaafb38 on 2024-04-04
LibMesh Version:         7dbd90ade88cb16a5e855a4c345454c9c146ebec
PETSc Version:           3.20.3
SLEPc Version:           3.20.1

**Reply by xajezade** [reply]
On a separate note, `modules/contact/examples/2d_indenter/` example cannot be restarted using `--recover` option. Also, I have outputted checkpoints. I tried to read from mesh.cpr file from input file but it gives me this error:`Cannot use appendMeshGenerator() because there is not a generator to append to!`

**Reply by GiudGiud** [reply]
@loganharbour for restart if you know what is going on with contact
@gambka @recuero for contact

**Reply by xajezade** [reply]
@schakrabortygithub @GiudGiud 
Here is a lighter version of my simulation that can reproduce the same problem that I am facing. Note that you have to let the simulation goes beyond 100 steps to make sure a contact is made.
[inputs_for_test.tar.gz](https://github.com/idaholab/moose/files/14953421/inputs_for_test.tar.gz)

I let it run to a step more than 100. Then, I use `--recover` option in CLI to recover my simulation and  the time increment decreases rapidly to a very small number while it was 0.01 stable time increment before termination.
